# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/glibc/glibc23/sparc-gdb_fix.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- glibc-2.3.2/sysdeps/unix/sysv/linux/sparc/sys/ucontext.h.orig	2003-08-03 19:45:12.000000000 +0000
+++ glibc-2.3.2/sysdeps/unix/sysv/linux/sparc/sys/ucontext.h	2003-08-03 19:47:29.000000000 +0000
@@ -19,6 +19,8 @@
 #ifndef _SYS_UCONTEXT_H
 #define _SYS_UCONTEXT_H	1
 
+#ifdef __GLIBC__
+
 #include <features.h>
 #include <signal.h>
 #include <bits/wordsize.h>
@@ -267,4 +269,5 @@
   } ucontext_t;
 
 #endif /* __WORDSIZE == 32 */
+#endif /* __GLIBC__ */
 #endif /* sys/ucontext.h */
